lib/model/weather/weather.dart

```diff
@@ -1,37 +1,22 @@
+import 'dart:convert';
 import 'package:flutter/material.dart';
 import 'package:flutter_training/common/result.dart';
+import 'package:flutter_training/model/weather/weather_forecast_target.dart';
+import 'package:flutter_training/model/weather/weather_info.dart';
 import 'package:yumemi_weather/yumemi_weather.dart';
 
-enum WeatherCondition {
-  sunny,
-  cloudy,
-  rainy,
-}
-
-extension _WeatherConditionExt on Iterable<WeatherCondition> {
-  WeatherCondition? byNameOrNull(String name) {
-    for (final element in this) {
-      if (element.name == name) {
-        return element;
-      }
-    }
-    return null;
-  }
-}
-
 class Weather {
   const Weather(this._weatherClient);
 
   final YumemiWeather _weatherClient;
 
-  Result<WeatherCondition?, String> fetchWeather() {
+  Result<WeatherInfo, String> fetchWeather(WeatherForecastTarget query) {
```

Comment:
@morikann 
[nits]
修正お願いします！

```suggestion
  Result<WeatherInfo, String> fetchWeather(WeatherForecastTarget target) {
```

Reply:
あ...修正忘れ申し訳ございません🙇‍♂️
こちらで修正しました。！
7a405843fa72ac2a1e3e7f70b87198f4991ff7fb